File: java-workshop/part-3-mocks-clean/src/main/java/domain/DiscountCode.java
package domain;

public enum DiscountCode {
    SUMMER20(20),
    ONCE50(50),
    WELCOME(10);

    private final int percentage;

    DiscountCode(int percentage) {
        this.percentage = percentage;
    }

    public int getPercentage() {
        return percentage;
    }
}

File: csharp-workshop/Part4.Behavior/Domain/DiscountCode.cs
namespace Part4.Behavior.Domain;

public enum DiscountCode
{
    SUMMER20,
    ONCE50,
    WELCOME
}

public static class DiscountCodeExtensions
{
    public static int GetPercentage(this DiscountCode code) => code switch
    {
        DiscountCode.SUMMER20 => 20,
        DiscountCode.ONCE50   => 50,
        DiscountCode.WELCOME  => 10,
        _ => throw new ArgumentOutOfRangeException(nameof(code), code, null)
    };
}
